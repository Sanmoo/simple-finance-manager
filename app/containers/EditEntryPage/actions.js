/*
 *
 * EditEntryPage actions
 *
 */

import {
  DEFAULT_ACTION,
  UPDATE_FORM_VALUE,
  SUBMIT_ENTRY,
  SAVE_KEY,
  ADD_NEW_ENTRY_FINISHED,
  CANCEL,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function updateFormValue(field, value) {
  return {
    type: UPDATE_FORM_VALUE,
    field,
    value,
  };
}

export function submitEntry(formValues) {
  return {
    type: SUBMIT_ENTRY,
    formValues,
  };
}

export function saveKey(key, value) {
  return {
    type: SAVE_KEY,
    key,
    value,
  };
}

export function addNewEntryFinished(params) {
  return {
    type: ADD_NEW_ENTRY_FINISHED,
    ...params,
  };
}

export function cancel() {
  return { type: CANCEL };
}
