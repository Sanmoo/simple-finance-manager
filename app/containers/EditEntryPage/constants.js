/*
 *
 * EditEntryPage constants
 *
 */

export const DEFAULT_ACTION = 'app/EditEntryPage/DEFAULT_ACTION';
export const UPDATE_FORM_VALUE = 'app/EditEntryPage/UPDATE_FORM_VALUE';
export const SUBMIT_ENTRY = 'app/EditEntryPage/SUBMIT_ENTRY';
export const SAVE_KEY = 'app/EditEntryPage/SAVE_KEY';
export const ADD_NEW_ENTRY_FINISHED =
  'app/EditEntryPage/ADD_NEW_ENTRY_FINISHED';
export const CANCEL = 'app/EditEntryPage/CANCEL';
