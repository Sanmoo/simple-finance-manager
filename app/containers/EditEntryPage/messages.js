/*
 * EditEntryPage Messages
 *
 * This contains all the text for the EditEntryPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.EditEntryPage';

export default defineMessages({
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Date',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description',
  },
  credit: {
    id: `${scope}.credit`,
    defaultMessage: 'Credit Card',
  },
  category: {
    id: `${scope}.category`,
    defaultMessage: 'Category',
  },
  value: {
    id: `${scope}.value`,
    defaultMessage: 'Value',
  },
});
