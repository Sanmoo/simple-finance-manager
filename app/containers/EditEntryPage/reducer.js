/*
 *
 * EditEntryPage reducer
 *
 */
import produce from 'immer';
import {
  DEFAULT_ACTION,
  UPDATE_FORM_VALUE,
  SAVE_KEY,
  ADD_NEW_ENTRY_FINISHED,
  SUBMIT_ENTRY,
  CANCEL,
} from './constants';

export const initialState = {
  formValues: {
    date: new Date(),
    description: '',
    category: '',
    credit: false,
    value: 0,
  },
  categories: [],
  submitInProgress: false,
  editingEntriesKey: null,
};

/* eslint-disable default-case, no-param-reassign, consistent-return */
const editEntryPageReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case DEFAULT_ACTION:
        break;
      case SUBMIT_ENTRY:
        draft.submitInProgress = true;
        break;
      case UPDATE_FORM_VALUE:
        draft.formValues[action.field] = action.value;
        break;
      case ADD_NEW_ENTRY_FINISHED:
        draft.submitInProgress = false;
        draft.formValues = action.success
          ? initialState.formValues
          : state.formValues;
        break;
      case CANCEL:
        return initialState;
      case SAVE_KEY:
        draft[action.key] = action.value;
        break;
    }
  });

export default editEntryPageReducer;
