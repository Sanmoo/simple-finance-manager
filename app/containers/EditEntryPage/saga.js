/**
 * Gets the repositories of the user from Github
 */

import { takeLatest, put } from 'redux-saga/effects';
import { addNewEntry } from 'utils/business';
import { popSnackbar } from 'containers/SnackbarContainer/actions';
import { push } from 'connected-react-router';
import { TYPE_EXPENSE } from 'utils/businessConstants';
import { addNewEntryFinished } from './actions';
import { SUBMIT_ENTRY, CANCEL } from './constants';

export function* handleSubmitEntry({ formValues }) {
  const editing = formValues.line;
  const { type } = formValues;
  const typeStr = type === TYPE_EXPENSE ? 'Despesa' : 'Receita';

  try {
    yield addNewEntry(formValues);
    yield put(addNewEntryFinished({ success: true }));
    // TODO remove hardcoded message from saga
    if (editing) {
      yield put(popSnackbar(`${typeStr} editada com sucesso`));
      yield put(push('/entries'));
    } else {
      yield put(popSnackbar(`${typeStr} adicionada com sucesso`));
      yield put(push('/'));
    }
  } catch (e) {
    yield put(addNewEntryFinished({ success: false }));
    // TODO remove console, alert and handle this error better
    /* eslint-disable no-alert, no-console */
    console.log(e);
    alert('There was an error while trying to save the new entry.');
  }
}

export function* handleCancel() {
  yield put(push('/'));
}

export default function* saga() {
  yield takeLatest(SUBMIT_ENTRY, handleSubmitEntry);
  yield takeLatest(CANCEL, handleCancel);
}
