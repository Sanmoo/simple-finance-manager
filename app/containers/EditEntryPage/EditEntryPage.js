/**
 *
 * EditEntryPage
 *
 */

import React, { useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Helmet } from 'react-helmet';
import { injectIntl, intlShape } from 'react-intl';
import MenuItem from '@material-ui/core/MenuItem';
import { withStyles } from '@material-ui/core/styles';
// import { FormattedMessage } from 'react-intl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import TextField from '@material-ui/core/TextField';
import Checkbox from '@material-ui/core/Checkbox';
import queryString from 'query-string';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';
import format from 'date-fns/format';
import parse from 'date-fns/parse';
import { modeFromLocation } from 'utils/routes';
import {
  loadExpenseCategoriesFromCurrentMonthCategoryGoals,
  loadIncomeCategoriesFromCurrentMonthCategoryGoals,
} from 'utils/business';
import CircularProgress from '@material-ui/core/CircularProgress';
import { TYPE_EXPENSE } from 'utils/businessConstants';
import messages from './messages';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: '100%',
  },
  dense: {
    marginTop: 19,
  },
  menu: {
    width: 200,
  },
  button: {
    margin: theme.spacing.unit,
  },
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
  },
  progress: {
    margin: theme.spacing.unit,
  },
});

export function EditEntryPage({
  classes,
  intl: { formatMessage },
  formValues: { date, description, credit, category, value },
  editingEntriesKey,
  formValues,
  updateFormValue,
  onCancelClick,
  onSubmit,
  categories,
  onCategoriesLoaded,
  spreadsheetId,
  submitInProgress,
  onSetupEditForm,
  location,
}) {
  const updateFormNumber = useCallback(
    evt => {
      const numberAsStr = evt.target.value;
      if (numberAsStr) {
        updateFormValue('value', parseFloat(numberAsStr));
      } else {
        updateFormValue('value', 0);
      }
    },
    [value, updateFormValue],
  );
  useEffect(() => {
    const mode = modeFromLocation(location);
    let promise;
    if (mode === TYPE_EXPENSE) {
      console.log('Confirmo que passei por aqui!');
      promise = loadExpenseCategoriesFromCurrentMonthCategoryGoals();
    } else {
      promise = loadIncomeCategoriesFromCurrentMonthCategoryGoals();
    }
    promise.then(onCategoriesLoaded);
  }, [location]);

  useEffect(() => {
    const editingEntry = location ? queryString.parse(location.search) : null;
    if (editingEntry && editingEntry.line) {
      onSetupEditForm(editingEntry);
    }
  }, [location]);

  const submitDisabled =
    !date || !description || !category || value === 0 || submitInProgress;

  const mode = modeFromLocation(location);

  const title =
    mode === TYPE_EXPENSE ? 'Adicionar Despesa' : 'Adicionar Receita';

  return (
    <div>
      <Helmet>
        <title>{title}</title>
        <meta
          name="description"
          content="Adicione uma nova entrada na planilha"
        />
      </Helmet>
      <form className={classes.container}>
        <TextField
          id="entry-date"
          label={formatMessage(messages.date)}
          className={classes.textField}
          type="date"
          value={format(date, 'yyyy-MM-dd')}
          onChange={evt =>
            updateFormValue(
              'date',
              parse(evt.target.value, 'yyyy-MM-dd', new Date()),
            )
          }
          margin="dense"
        />
        <TextField
          id="entry-description"
          label={formatMessage(messages.description)}
          className={classes.textField}
          value={description}
          multiline
          onChange={evt => updateFormValue('description', evt.target.value)}
          margin="dense"
        />
        {mode === TYPE_EXPENSE && (
          <FormControlLabel
            control={
              <Checkbox
                checked={credit}
                onChange={evt => updateFormValue('credit', evt.target.checked)}
                color="primary"
              />
            }
            className={classes.textField}
            label={formatMessage(messages.credit)}
          />
        )}
        <TextField
          id="entry-category"
          select
          label={formatMessage(messages.category)}
          className={classes.textField}
          value={category}
          onChange={evt => updateFormValue('category', evt.target.value)}
          SelectProps={{
            MenuProps: {
              className: classes.menu,
            },
          }}
          margin="normal"
        >
          {categories.map(categoryOption => (
            <MenuItem key={categoryOption} value={categoryOption}>
              {categoryOption}
            </MenuItem>
          ))}
        </TextField>
        <TextField
          id="entry-value"
          label={formatMessage(messages.value)}
          className={classes.textField}
          value={value || ''}
          type="number"
          onChange={updateFormNumber}
          margin="dense"
          InputLabelProps={{
            shrink: true,
          }}
        />
      </form>
      {/* Form Actions */}
      <div>
        <Button
          variant="contained"
          size="small"
          className={classes.button}
          onClick={() =>
            onSubmit({
              ...formValues,
              ...(editingEntriesKey || {}),
              type: mode,
              spreadsheetId,
            })
          }
          disabled={submitDisabled}
        >
          {!submitInProgress && (
            <SaveIcon
              className={classNames(classes.leftIcon, classes.iconSmall)}
            />
          )}
          {submitInProgress && (
            <CircularProgress
              className={classNames(classes.leftIcon, classes.iconSmall)}
              size={20}
            />
          )}
          Save
        </Button>
        <Button
          variant="contained"
          size="small"
          className={classes.button}
          onClick={onCancelClick}
          disabled={submitInProgress}
        >
          Cancel
        </Button>
      </div>
    </div>
  );
}

EditEntryPage.propTypes = {
  classes: PropTypes.object.isRequired,
  intl: intlShape.isRequired,
  formValues: PropTypes.shape({
    date: PropTypes.instanceOf(Date).isRequired,
    description: PropTypes.string.isRequired,
    credit: PropTypes.bool.isRequired,
    category: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
  }),
  updateFormValue: PropTypes.func.isRequired,
  onCancelClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  onCategoriesLoaded: PropTypes.func.isRequired,
  spreadsheetId: PropTypes.string.isRequired,
  submitInProgress: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }),
  onSetupEditForm: PropTypes.func.isRequired,
  editingEntriesKey: PropTypes.shape({
    line: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    originSheetTitle: PropTypes.string.isRequired,
  }),
};

export default withStyles(styles)(injectIntl(EditEntryPage));
