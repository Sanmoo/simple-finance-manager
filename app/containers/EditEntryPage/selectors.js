import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the editEntryPage state domain
 */

const selectEditEntryPageDomain = state => state.editEntryPage || initialState;

/**
 * Other specific selectors
 */

/**
 * Default selector used by EditEntryPage
 */

const makeSelectEditEntryPage = () =>
  createSelector(
    selectEditEntryPageDomain,
    substate => substate,
  );

const makeSelectFormValues = () =>
  createSelector(
    makeSelectEditEntryPage(),
    substate => substate.formValues,
  );

const makeSelectCategories = () =>
  createSelector(
    makeSelectEditEntryPage(),
    substate => substate.categories,
  );

const makeSelectSubmitInProgress = () =>
  createSelector(
    makeSelectEditEntryPage(),
    substate => substate.submitInProgress,
  );

const makeSelectEditingEntriesKey = () =>
  createSelector(
    makeSelectEditEntryPage(),
    substate => substate.editingEntriesKey,
  );

export default makeSelectEditEntryPage;
export {
  selectEditEntryPageDomain,
  makeSelectFormValues,
  makeSelectCategories,
  makeSelectSubmitInProgress,
  makeSelectEditingEntriesKey,
};
