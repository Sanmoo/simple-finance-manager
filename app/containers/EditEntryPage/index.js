/**
 *
 * EditEntryPage
 *
 */

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import { makeSelectSId } from 'containers/App/selectors';

import {
  makeSelectFormValues,
  makeSelectCategories,
  makeSelectSubmitInProgress,
} from './selectors';
import reducer from './reducer';
import saga from './saga';
import { updateFormValue, submitEntry, saveKey, cancel } from './actions';
import EditEntryPage from './EditEntryPage';

const mapStateToProps = createStructuredSelector({
  formValues: makeSelectFormValues(),
  categories: makeSelectCategories(),
  spreadsheetId: makeSelectSId(),
  submitInProgress: makeSelectSubmitInProgress(),
});

export function mapDispatchToProps(dispatch) {
  return {
    updateFormValue: (field, value) => dispatch(updateFormValue(field, value)),
    onCancelClick: () => dispatch(cancel()),
    onSubmit: formValues => dispatch(submitEntry(formValues)),
    onCategoriesLoaded: categories =>
      dispatch(saveKey('categories', categories)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'editEntryPage', reducer });
const withSaga = injectSaga({ key: 'editEntryPage', saga });

export default compose(
  withConnect,
  withReducer,
  withSaga,
)(EditEntryPage);
