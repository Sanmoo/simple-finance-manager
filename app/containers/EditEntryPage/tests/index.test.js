/**
 *
 * Tests for EditEntryPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { EditEntryPage } from '../EditEntryPage';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<EditEntryPage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EditEntryPage dispatch={dispatch} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EditEntryPage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
