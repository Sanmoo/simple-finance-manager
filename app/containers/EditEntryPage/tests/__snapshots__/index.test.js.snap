// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditEntryPage /> Should render and match the snapshot 1`] = `
<div>
  <form
    class="InjectIntl-EditEntryPage--container-120"
  >
    <div
      class="MuiFormControl-root-128 MuiFormControl-marginDense-130 InjectIntl-EditEntryPage--textField-121"
    >
      <label
        class="MuiFormLabel-root-143 MuiFormLabel-filled-147 MuiInputLabel-root-132 MuiInputLabel-formControl-137 MuiInputLabel-animated-140 MuiInputLabel-shrink-139 MuiInputLabel-marginDense-138"
        data-shrink="true"
        for="entry-date"
      >
        Date
      </label>
      <div
        class="MuiInputBase-root-163 MuiInput-root-150 MuiInput-underline-154 MuiInputBase-formControl-164 MuiInput-formControl-151 MuiInputBase-marginDense-170"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-173 MuiInput-input-158 MuiInputBase-inputType-176 MuiInput-inputType-161 MuiInputBase-inputMarginDense-174 MuiInput-inputMarginDense-159"
          id="entry-date"
          type="date"
          value="2019-06-10"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-128 MuiFormControl-marginDense-130 InjectIntl-EditEntryPage--textField-121"
    >
      <label
        class="MuiFormLabel-root-143 MuiFormLabel-filled-147 MuiInputLabel-root-132 MuiInputLabel-formControl-137 MuiInputLabel-animated-140 MuiInputLabel-shrink-139 MuiInputLabel-marginDense-138"
        data-shrink="true"
        for="entry-description"
      >
        Description
      </label>
      <div
        class="MuiInputBase-root-163 MuiInput-root-150 MuiInput-underline-154 MuiInputBase-formControl-164 MuiInput-formControl-151 MuiInputBase-marginDense-170 MuiInputBase-multiline-171 MuiInput-multiline-156"
      >
        <div
          class="MuiPrivateTextarea-root-180"
        >
          <textarea
            aria-hidden="true"
            class="MuiPrivateTextarea-textarea-181 MuiPrivateTextarea-shadow-182"
            readonly=""
            rows="1"
            tabindex="-1"
          />
          <textarea
            aria-hidden="true"
            class="MuiPrivateTextarea-textarea-181 MuiPrivateTextarea-shadow-182"
            readonly=""
            rows="1"
            tabindex="-1"
          >
            anything
          </textarea>
          <textarea
            aria-invalid="false"
            class="MuiPrivateTextarea-textarea-181 MuiInputBase-input-173 MuiInput-input-158 MuiInputBase-inputMultiline-175 MuiInput-inputMultiline-160 MuiInputBase-inputMarginDense-174 MuiInput-inputMarginDense-159"
            id="entry-description"
            rows="1"
            style="height: 19px;"
          >
            anything
          </textarea>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root-128 MuiFormControl-marginNormal-129 InjectIntl-EditEntryPage--textField-121"
    >
      <label
        class="MuiFormLabel-root-143 MuiFormLabel-filled-147 MuiInputLabel-root-132 MuiInputLabel-formControl-137 MuiInputLabel-animated-140 MuiInputLabel-shrink-139"
        data-shrink="true"
        for="entry-category"
      >
        Category
      </label>
      <div
        class="MuiInputBase-root-163 MuiInput-root-150 MuiInput-underline-154 MuiInputBase-formControl-164 MuiInput-formControl-151"
      >
        <div
          class="MuiSelect-root-183"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            class="MuiSelect-select-184 MuiSelect-selectMenu-187 MuiInputBase-input-173 MuiInput-input-158"
            role="button"
            tabindex="0"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            id="entry-category"
            type="hidden"
            value="My Category"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-190 MuiSelect-icon-189"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root-128 MuiFormControl-marginDense-130 InjectIntl-EditEntryPage--textField-121"
    >
      <label
        class="MuiFormLabel-root-143 MuiFormLabel-filled-147 MuiInputLabel-root-132 MuiInputLabel-formControl-137 MuiInputLabel-animated-140 MuiInputLabel-shrink-139 MuiInputLabel-marginDense-138"
        data-shrink="true"
        for="entry-value"
      >
        Value
      </label>
      <div
        class="MuiInputBase-root-163 MuiInput-root-150 MuiInput-underline-154 MuiInputBase-formControl-164 MuiInput-formControl-151 MuiInputBase-marginDense-170"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-173 MuiInput-input-158 MuiInputBase-inputType-176 MuiInput-inputType-161 MuiInputBase-inputMarginDense-174 MuiInput-inputMarginDense-159"
          id="entry-value"
          type="number"
          value="50"
        />
      </div>
    </div>
  </form>
  <div>
    <button
      class="MuiButtonBase-root-229 MuiButton-root-203 MuiButton-contained-214 MuiButton-raised-217 MuiButton-sizeSmall-226 InjectIntl-EditEntryPage--button-124"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-204"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-190 InjectIntl-EditEntryPage--leftIcon-125"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
          />
        </svg>
        Save
      </span>
      <span
        class="MuiTouchRipple-root-232"
      />
    </button>
    <button
      class="MuiButtonBase-root-229 MuiButton-root-203 MuiButton-contained-214 MuiButton-raised-217 MuiButton-sizeSmall-226 InjectIntl-EditEntryPage--button-124"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-204"
      >
        Cancel
      </span>
      <span
        class="MuiTouchRipple-root-232"
      />
    </button>
  </div>
</div>
`;
